internal/msgqueue/v1/msgqueue.go

```diff
@@ -0,0 +1,162 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hatchet-dev/hatchet/pkg/random"
+)
+
+type Queue interface {
+	// Name returns the name of the queue.
+	Name() string
+
+	// Durable returns true if this queue should survive task queue restarts.
+	Durable() bool
+
+	// AutoDeleted returns true if this queue should be deleted when the last consumer unsubscribes.
+	AutoDeleted() bool
+
+	// Exclusive returns true if this queue should only be accessed by the current connection.
+	Exclusive() bool
+
+	// FanoutExchangeKey returns which exchange the queue should be subscribed to. This is only currently relevant
+	// to tenant pub/sub queues.
+	//
+	// In RabbitMQ terminology, the existence of a subscriber key means that the queue is bound to a fanout
+	// exchange, and a new random queue is generated for each connection when connections are retried.
+	FanoutExchangeKey() string
+
+	// DLX returns the dead letter exchange for the queue, if it exists.
+	DLX() string
+}
+
+type staticQueue string
+
+const (
+	EVENT_PROCESSING_QUEUE    staticQueue = "event_processing_queue_v2"
+	JOB_PROCESSING_QUEUE      staticQueue = "job_processing_queue_v2"
+	WORKFLOW_PROCESSING_QUEUE staticQueue = "workflow_processing_queue_v2"
+
+	TASK_PROCESSING_QUEUE staticQueue = "task_processing_queue_v2"
+	OLAP_QUEUE            staticQueue = "olap_queue_v2"
```

Comment:
nit: should these be `v1` or `v2`?

Reply:
we can't rename the old ones without dropping some events during a redeploy, and I think it'd be confusing to call the new ones with a lower version number?

Reply:
womp womp, yeah, makes sense to me 👍